newton/_src/solvers/mujoco/solver_mujoco.py

```
@@ -1758,6 +1950,9 @@ def add_geoms(newton_body_id: int):
             mb1, mb2 = body_mapping[b1], body_mapping[b2]
             spec.add_exclude(bodyname1=spec.bodies[mb1].name, bodyname2=spec.bodies[mb2].name)
 
+        # add explicit contact pairs from custom attributes
+        self._init_pairs(model, spec, shape_mapping, first_group)
```

Ruff (0.14.10)

Undefined name `first_group`

(F821)